content/articles/2018-06-bench-1.0.1.Rmd

```diff
@@ -0,0 +1,190 @@
+---
+title: bench 1.0.1
+date: '2018-06-28'
+slug: bench-1.0.1
+author: Jim Hester
+categories: [package]
+description: >
+  bench 1.0.1 is now available on CRAN. bench allows you to benchmark
+  code, tracking execution time, memory allocations and garbage
+  collections.
+photo:
+  url: https://unsplash.com/photos/nIkuMWT4Imc
+  author: Rula Sibai
+---
+
+[bench] is now available on CRAN!
+
+The goal of [bench] is to benchmark code, tracking execution time,
+memory allocations and garbage collections.
+
+Install the latest version with:
+``` r
+install.packages("bench")
+```
+
+### Comparison vs existing methods
+
+Compared to other methods such as [system.time], [rbenchmark], [tictoc] or
+[microbenchmark] we feel it has a number of benefits.
+
+- Always uses the highest precision APIs available for each operating system (often nanosecond-level).
+- Tracks memory allocations for each expression benchmarked.
+- Tracks the number and type of R garbage collections per expression iteration.
+- Verifies equality of expression results by default, to avoid accidentally
+  benchmarking inequivalent code.
+- `bench::press()`, a function which allows you to easily perform and combine
+  benchmarks across a large grid of values.
+- Uses adaptive stopping by default, running each expression for a set amount
+  of time rather than for a specific number of iterations.
+- Expressions are run in batches and summary statistics are calculated after
+  filtering out iterations with garbage collections. This allows you to isolate
+  the performance  and effects of garbage collection on running time (for more details see [Neal
+  2014](https://radfordneal.https://dynverse.github.io/dynverse/wordpress.com/2014/02/02/inaccurate-results-from-microbenchmark/)).
+
+### Usage
+
+Benchmarks can be run with `bench::mark()`, which takes one or more expressions
+to benchmark against each other.
+
+```{r example}
+library(bench)
+set.seed(42)
+dat <- data.frame(x = runif(10000, 1, 1000), y=runif(10000, 1, 1000))
+```
+
+`bench::mark()` will throw an error if the results are not equivalent, so you
+don't accidentally benchmark inequivalent code.
+```{r example1-1, error = TRUE}
+bench::mark(
+  dat[dat$x > 500, ],
+  dat[which(dat$x > 499), ],
+  subset(dat, x > 500))
+```
+
+Results are easy to interpret, with human readable units.
+```{r example1-2}
+bnch <- bench::mark(
+  dat[dat$x > 500, ],
+  dat[which(dat$x > 500), ],
+  subset(dat, x > 500))
+bnch
+```
+
+By default the summary uses absolute measures, however relative results can be
+obtained by using `relative = TRUE` in your call to `bench::mark()` or calling
+`summary(relative = TRUE)` on the results.
```

Comment:
I might add a second by before "calling," but that's dealer's choice stylistically.